src/contracts/AccessList.ts

```diff
@@ -122,10 +124,11 @@ export class AccessListContract extends SmartContractWithAddress {
   ): Promise<ReceiptOrEstimate<G>> {
     const estGas = await this.contract.estimateGas.batchMint(users, tokenUris)
     if (estimateGas) return <ReceiptOrEstimate<G>>estGas
-
     const trxReceipt = await sendTx(
       estGas,
-      this.signer,
+      this.config && 'sdk' in this.config && this.config.sdk === 'oasis'
```

Comment:
I would create a function for this check... we're repeating the same lines of code multiple times. Also, since we don't have a constants or a enum type... its better to not write the same strings in multiple places, its very error prone

Reply:
I agree that we need to remove the code duplication from the contract level and put it into sendTx. What is unclear for me is that how we can manage to retrieve the config if it was already defined by the user.
If the user already defined it's own config object, we have to retrieve it without creating a new one. Should we have a check in the 'args' or set an optional parameter for config within `sendTx`? I am saying even I will create a separate function, we still need to verify if there is an existing config object already set.

Reply:
> I agree that we need to remove the code duplication from the contract level and put it into sendTx. What is unclear for me is that how we can manage to retrieve the config if it was already defined by the user. If the user already defined it's own config object, we have to retrieve it without creating a new one. Should we have a check in the 'args' or set an optional parameter for config within `sendTx`? I am saying even I will create a separate function, we still need to verify if there is an existing config object already set.

Not sure what's the problem, if the config was passed as argument to the smart contract class?? .
The Config is a collection of 'network' configurations... and for each network there is only one entry, so if you do a ` new ConfigHelper().getConfig(network)` i don't see how that could give a different entry from the "original one" (if any)..
Afaik, the "args" are the params for the smart function to be called... putting the config there not only would force the user to pass it, but it would be even more messy, since that is not a SC function argument... Thats really confusing...

Reply:
there are possibilities as well. You could have a method on abstract `SmartContract` class, like
getWrappedSigner() or whatever
where you could put the repeated code:
```
this.config && 'sdk' in this.config && this.config.sdk === 'oasis'
        ? sapphire.wrap(this.signer)
        : this.signer,
```
that returns the Signer (wrapped or not)

Then inside each class where `sendTx() `is beeing called you could just put `this.getWrappedSigner()` instead of those lines.. In this case we would not touch `sendTx` ... it would still be 93 changes but much cleaner

Reply:
I prefer the second method it will be much cleaner since we'll call a function, not having duplicated lines of code.